layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -1088,14 +1085,31 @@ Decision is based on `dotspacemacs-line-numbers'."
                (* 1000 (car (spacemacs/mplist-get dotspacemacs-line-numbers
                                                   :size-limit-kb)))))))
 
+;; mode in :enabled, not in :disabled ==> t
+;; mode not in :enabled, in :disabled ==> nil
+;; mode in :enabled, parent in :disabled ==> t
+;; parent in :enabled, mode in :disabled ==> nil
+;; not in :enabled, not in :disabled, :enabled is empty ==> t
+;; not in :enabled, not in :disabled, :enabled is not empty ==> nil
+;; both :enabled and :disabled are empty ==> t
 (defun spacemacs//linum-enabled-for-current-major-mode ()
   "Return non-nil if line number is enabled for current major-mode."
-  (let ((modes (spacemacs/mplist-get dotspacemacs-line-numbers
-                                     :enabled-for-modes)))
-    (memq major-mode modes)))
-
-(defun spacemacs//linum-disabled-for-current-major-mode ()
-  "Return non-nil if line number is disabled for current major-mode."
-  (let ((modes (spacemacs/mplist-get dotspacemacs-line-numbers
-                                     :disabled-for-modes)))
-    (memq major-mode modes)))
+  (let* ((enabled-for-modes (spacemacs/mplist-get dotspacemacs-line-numbers
+                                                  :enabled-for-modes))
+         (disabled-for-modes (spacemacs/mplist-get dotspacemacs-line-numbers
+                                                   :disabled-for-modes))
+         (enabled-for-parent (apply #'derived-mode-p enabled-for-modes))
+         (disabled-for-parent (apply #'derived-mode-p disabled-for-modes))
+         (enabled-derives-from-disabled-p (and enabled-for-parent
```

Comment:
I still didn't really understand this name or the exact purpose of that binding. Maybe we could shorten it, only keep the `let` form there and put the first part of the `and` down there with the rest ?

I think it would make it more readable.